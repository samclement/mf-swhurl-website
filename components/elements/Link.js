import React from 'react'
import Link from 'next/link';
import styled from 'styled-components';

const StyledLink = styled.a`
  color: 'deeppink';
`

export default ({href, name}) => (
  <Link prefetch href={href} passHref>
    <StyledLink>{name}</StyledLink>
  </Link>
)
