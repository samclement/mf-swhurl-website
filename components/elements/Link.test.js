import React from 'react'
import { render } from 'enzyme'
import 'jest-styled-components'

import Link from './Link.js'

describe('With Snapshot Testing', () => {
  it('Link shows "About" with href "/about"', () => {
    const wrapper = render(<Link href="/about" name="About" />)
    expect(wrapper).toMatchSnapshot()
  })
})
