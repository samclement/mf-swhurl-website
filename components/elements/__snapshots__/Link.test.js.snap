// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Link shows "About" with href "/about" 1`] = `
.c0 {
  color: 'deeppink';
}

<a
  class="c0"
  href="/about"
>
  About
</a>
`;
